---
title: "Code review: TET2 and hypermethylation"
author: "Tim Triche"
date: "November 22nd, 2021"
output: 
  html_document:
    keep_md: true
vignette: >
  %\VignetteIndexEntry{TET2}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---
I needed another package installed before this code would run. Package & Code is in the below chunk
```{r}
BiocManager::install('GEOquery')
```


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
library(devtools)
load_all("./")
```

# Installation

Install the WorldsSimplestCodeReview package, if you haven't. 
#This chunk was not necessary to run for me.-Riley

```{r, loadpkgs, eval = FALSE, message = FALSE}
#install.packages("remotes")
#install.packages("BiocManager")
#BiocManager::install("VanAndelInstitute/WorldsSimplestCodeReview")
library(knitr)
```

To extract just the R code, you can use knitr::knit(input, tangle=TRUE):
#This chunk was not necessary to run for me.-Riley
```{r, tangle, eval = FALSE, message = FALSE, echo = FALSE}
# knitr::knit("TET2.Rmd", tangle = TRUE) 
# [1] "TET2.R"
```

# Introduction

Long before any of you were born, back in 2010, an exciting paper came out 
which purported to show that _IDH1_, _IDH2_, and _TET2_ mutations shared a 
phenotype of hypermethylation owing to loss of 5-hydroxymethylcytosine. The 
details can be found in [the paper](https://doi.org/10.1016/j.ccr.2010.11.015), 
which is indeed a landmark. Nevertheless, some fine details of the work seemed
to disagree with the results of other cohorts when replication was attempted.

![The money shot](figure/TET2.png)

Some of you who have seen volcano plots before can guess where this is going.

# The data
Loading the data from the paper. This requred loading packages limma and GEOquery
```{r, fetchGEO}
#Load a package that can be used for Data analysis, linear models, and differential expression for microarray data. This is part of the BioConductor package.
library(limma)
#Load GEOquery package, that allows you to get data from the NCBI Gene Expression Omnibus
library(GEOquery)
#Load the paper's dataset specifically, if it isn't already; but just the parts you want
if (!exists("DNAme")) data(DNAme)

if (FALSE) { # this takes about 5 minutes:

  # needed to fetch data
  library(GEOquery) 
  #ID for paper is GSE24505 make a new dataset called MSK_HOVON
  #MSK_HOVON <- getGEO("GSE24505") returned the error that the "The size of the connection buffer (131072)        was not large enough to fit a complete line"
  #increase vroom connection size x100 allowed that line of code to work
  Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 100)
  MSK_HOVON <- getGEO("GSE24505")

  # skip the expression data:
  
  #make a platform that can easily be subsetted by annotation
  platform <- sapply(MSK_HOVON, annotation)
  #Define methylation data as an extractable value from platform
  methylation <- which(platform == "GPL6604")
 
   #Get only methylation from MSK_HOVON
  DNAme <- MSK_HOVON[[methylation]] # GPL6604, HG17_HELP_PROMOTER 
    #Get sex variable from dataset
  DNAme$male <-ifelse(DNAme$characteristics_ch1=="sex (male.1_female.2): 1",1,0)
    #Get TET2 methylation variable from dataset
  DNAme$TET2 <- ifelse(DNAme$characteristics_ch1.7 == "tet2: WT", 0, 1)
    #Get IDH methylation variable from dataset
  DNAme$IDH <- ifelse(DNAme$characteristics_ch1.8 == "idh1.idh2: WT", 0, 1)
    #Get Purity variable from dataset
  DNAme$purity <- as.integer(DNAme$"bm_%blasts:ch1") / 100
    #Save this file! So that next time: if (!exists("DNAme")) data(DNAme) : will return "TRUE"
  save(DNAme, file="../data/DNAme.rda")

}

# how many probes, how many patients?
dim(DNAme)
# Features  Samples
#    25626      394

#SO, we have 25626 probes and 394 patients/samples
```

### Some contrasts

Is it the case that TET2, IDH1, and IDH2 mutations are exclusive?

```{r, heatmap, eval=TRUE}

# always plot your data
#this is the type of plot we are using...
library(ComplexHeatmap)

#make a matrix of methylation data, coded as a binary
mutations <- t(as.matrix(pData(DNAme)[, c("TET2", "IDH")]))
#plot the heatmap, I decreased font size to 4 so that single double mutant name was visible
  #I like the color purple better than red
Heatmap(mutations, col=c("lightgray","purple"), name="mutant", column_km=4,row_km = 1,
        column_names_gp = gpar(fontsize = 4))

#Group 1 has no mutations, 2 has only IDH muations, 3 has only TET2 mutations, 4 has both TET2 and IDH mutations
```

Do we see genome-wide hypermethylation from TET2 mutations? 

```{r, TET2_vs_IDH}

# model TET2 and IDH1/2 mutant related hypermethylation
# note: there are plenty of confounders (pb%, bm%, wbc) that could be included
library(limma) 

# simplest design
design1 <- with(pData(DNAme), model.matrix( ~ IDH + TET2 ))
fit1 <- eBayes(lmFit(exprs(DNAme), design1))
(IDH_diffmeth_probes_fit1 <- nrow(topTable(fit1, 
                                           coef=grep("IDH", colnames(design1)), 
                                           p.value=0.05, # change if you like 
                                           number=Inf)))
# 6513 probes for IDH

(TET_diffmeth_probes_fit1 <- nrow(topTable(fit1, 
                                           coef=grep("TET2", colnames(design1)),
                                           p.value=0.05, # change if you like 
                                           number=Inf)))
# 6 probes for TET2

# control for sex
design2 <- with(pData(DNAme), model.matrix( ~ IDH + TET2 + male ))
fit2 <- eBayes(lmFit(exprs(DNAme), design2))
(IDH_diffmeth_probes_fit2 <- nrow(topTable(fit2, 
                                           coef=grep("IDH", colnames(design2)), 
                                           p.value=0.05, # change if you like 
                                           number=Inf)))
# 6651 probes for IDH 

(TET2_diffmeth_probes_fit2 <- nrow(topTable(fit2, 
                                            coef=grep("TET", colnames(design2)),
                                            p.value=0.05, # change if you like 
                                            number=Inf)))
# 7 probes for TET2

# control for blast count
design3 <- with(pData(DNAme), model.matrix( ~ IDH:purity + TET2:purity))
fit3 <- eBayes(lmFit(exprs(DNAme)[, as.integer(rownames(design3))], design3))

(IDH_diffmeth_probes_fit3 <- nrow(topTable(fit3, 
                                           coef=grep("IDH", colnames(design3)), 
                                           p.value=0.05, # change if you like 
                                           number=Inf)))
# 7450 probes for IDH:purity

(TET2_diffmeth_probes_fit3 <- nrow(topTable(fit3, 
                                            coef=grep("TET", colnames(design3)),
                                            p.value=0.05, # change if you like 
                                            number=Inf)))
# 10 probes for TET2:purity

```

In sum, there are more probes for IDH than TET2, on the order of 1000x more probes no matter which design (controlling for sex, purity, etc.)
If this is the case, then the conclusions of this paper (thatboth mutations are associated with a hypermethylation pattern) should be called into question.
Instead, it appears that the methylation patterns are different between TET2 vs IDH
